### ISSUE TITLE
Drop Waf build system support

### ISSUE THREAD

**Issue by eht16** [geany/geany#768]
As discussed in various discussions in the past, e.g. #679 and #524, we finally remove Waf build system support from Geany, mainly to reduce the build system maintenance overhead.

Steps to do:
- [x] Remove code references in GIT master, including manual (#769)
- [x] Remove website references
- [x] Remove wiki references

More?

**Comment by eht16** [comment]
Website references: done (sort of).
I slightly updated https://www.geany.org/Support/RunningOnWindows to mention (parts of) MSYS2 is necessary.
I changed the links to the very outdated pages https://www.geany.org/Support/BuildingOnWin32 and https://www.geany.org/Support/CrossCompile to the Wiki.

So the website is more or less done. I will update the webpages again or restructure the information once the website is ready.

Wiki is WIP.

**Comment by eht16** [comment]
Wiki status:
  - https://wiki.geany.org/howtos/win32/build The old Waf howto -> I added big warning about deprecated build system
  - https://wiki.geany.org/howtos/win32/crosscompile Cross-Compiling howto -> needs heavy update and rewrite, this is copied from the website and very outdated. I have no recent experiences yet with cross-compiling
  - https://wiki.geany.org/howtos/win32/msys2 Current MSYS2 build howto -> I added information about bundle and installer creation as used for 1.27. The steps are not nice and not very clean but it is a start.
  - https://wiki.geany.org/howtos/win32/running Runtime information -> needs update for MSYS2
